### DISCUSSION TITLE
Possible Julia front-end?

### DISCUSSION THREAD

**Discussion by xuhao1** [taichi-dev/taichi#4849] [Ideas]
Hi there,
As discussed during Taichi's open office hour, taichi is planning to develop more front-end(JS, Rust, etc.) Personally, I think a possible frontend is Julia:
1. Julia is a high efficiency (much better than Python) language and has a good community on high-performance computing
2. Julia is based on LLVM. I am not familiar with this area but looks like it will make development easier.
3. Currently, Julia also has CUDA packages, but they are not as convenient as taichi.

**Comment by k-ye** [comment]
We haven't really investigated much into Julia so far, but this definitely sounds like a great idea! Our current plan is to have a language-neutral AST layer, then it shouldn't be too hard to transpile a subset of any sane language to Taichi in theory. @AmesingFlank and @devillove084 might have more input on this matter?
